pkg/safetime/safetime.go

```diff
@@ -16,24 +15,24 @@ import (
 // returns false to indicate the fact.
 //
 // Used to workaround a malfunctioning monotonic clock.
-func TimeSinceSafe(t time.Time, logger *logrus.Entry) (time.Duration, bool) {
+func TimeSinceSafe(t time.Time, logger *slog.Logger) (time.Duration, bool) {
```

Comment:
Offtopic, but is this package still relevant? Most of the codebase simply does `time.Since`, so we could probably just get rid of it.

Reply:
@giorio94 Yep, I just read the commit description that added this and I remember this happening:
```
The wrapper "TimeSinceSafe" is used to work around the case when CLOCK_MONOTONIC
returns a non-monotonic time (observed on a burstable EC2 instance), which
results in "time.Since" returning a negative duration.

Passing a negative value to some Prometheus metrics methods (e.g. "Counter.Add")
can cause a panic. Thus, for the sake of safety, the wrapper should be used
instead "time.Since" when calling the methods.

Fixes #6222
```